src/coreclr/src/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -1357,6 +1357,11 @@ private uint getClassSize(CORINFO_CLASS_STRUCT_* cls)
                 throw new RequiresRuntimeJitException(type);
             }
 #endif
+            if (type.IsGenericDefinition)
```

Comment:
Would be better to prevent computing the layout for uninstantiated types instead (so that it throws in `GetElementSize` a couple lines above). Adding the check to `MetadataFieldLayoutAlgorithm.cs` might be better - I would place it somewhere where we do the other `ThrowHelper.ThrowTypeLoadException` in `ComputeInstanceLayout`.

(Also note this is TypeLayoutException, not InvalidProgramException. It will matter more when/if we start pregenerating throwing method bodies for these - it already matters for CoreRT.)

Reply:
Thank you for the suggestion, I'll give it a try.